diff --git a/hw/mips/malta.c b/hw/mips/malta.c
index 8e9cea70b1..2268a8b618 100644
--- a/hw/mips/malta.c
+++ b/hw/mips/malta.c
@@ -65,7 +65,7 @@
 #define ENVP_PADDR          0x2000
 #define ENVP_VADDR          cpu_mips_phys_to_kseg0(NULL, ENVP_PADDR)
 #define ENVP_NB_ENTRIES     16
-#define ENVP_ENTRY_SIZE     256
+#define ENVP_ENTRY_SIZE     1024
 
 /* Hardware addresses */
 #define FLASH_ADDRESS       0x1e000000ULL
