diff --git a/coregrind/m_signals.c b/coregrind/m_signals.c
index 17069df41..84b3de81e 100644
--- a/coregrind/m_signals.c
+++ b/coregrind/m_signals.c
@@ -1203,8 +1203,10 @@ static void handle_SCSS_change ( Bool force_update )
             VG_(sigfillset)( &ksa_old.sa_mask );
          }
 #        endif
+#        if !defined(VGP_mips32_linux) && !defined(VGP_mips64_linux)
          vg_assert(ksa_old.sa_flags 
                    == skss_old.skss_per_sig[sig].skss_flags);
+#        endif
 #        if !defined(VGP_ppc32_linux) && !defined(VGP_mips32_linux) && \
             !defined(VGP_mips64_linux) && !defined(VGP_nanomips_linux) && \
             !defined(VGP_riscv64_linux) && \
